### ISSUE TITLE
Win_firewall is not working with ansible  2.0.0.2

### ISSUE THREAD

**Issue by badial** [ansible/ansible-modules-extras#2624]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

win_firewall_rule
##### ANSIBLE VERSION

```
ansible 2.0.0.2
```
##### SUMMARY

Win_firewall is not working with ansible 2.0.0.2
##### ACTUAL RESULTS

```

An exception occurred during task execution. To see the full traceback, use -vvv. The error was: + ~~~~~~~~~~~~~~~~~~~~~~~
fatal: [52.77.237.246]: FAILED! => {"changed": false, "failed": true, "msg": "The property 'failed' cannot be found on this object. Verify that the property exists."}                                                                              


```

**Comment by ansibot** [comment]
@TimothyVandenbrande ping, this issue is waiting for your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by t-v** [comment]
I believe this issue has also been resolved_by_pr #2432
Otherwise, for extra testing, could you give me the task you were trying to run?
needs_info

**Comment by trondhindenes** [comment]
I'm also seeing an issue with the current version of this module. 
I'm getting an error that the property "service" doesn't exist on the object. This seems related to the parsed output of netsh advfirewall, where there is no property called "service", while the compare object has this (set to any, which is coming from the parsed ansible module input). Can be tested with:

```
- name: Firewall rule for logstash IIS (6001)
  action: win_firewall_rule
  args:
      name: Logstash_6001
      enable: yes
      state: present
      localport: 6001
      action: allow
      direction: In
      protocol: TCP
      profile: any
      force: True
  notify: restart logstash-iis
```

In my case, it fails when the rule already exists.

**Comment by dreamcat4** [comment]
Hello again.

Experiencing another issue with this module on windows 10... The module reports back 'Created firewall rule...' but no such rule exists. It makes me wonder if anyone else is seeing a similar thing happen to them?

**Comment by dreamcat4** [comment]
Another thing about this module: It uses the `netsh` command. I am assuming thats for some compatibility reasons? The thing is that today I just came across these handy powershell commands:

```
Get-Netfirewallrule
Enable-NetFirewallRule
Disable-Netfirewallrule
etc...
```

Have just tried them... they work pretty awesome on windows 10! Perhaps if have other older versions of windows can also try them too, see if they work ?

EDIT: Anyway here some more info about them

https://letitknow.wordpress.com/2012/09/13/manage-windows-firewall-with-powershell-3/

http://www.windowsecurity.com/articles-tutorials/Windows_Server_2012_Security/use-windows-command-line-tools-and-powershell-cmdlets-manage-security-windows-server-2012-part3.html

**Comment by t-v** [comment]
they don't work in win2003 and I thought not even in 2008 (not the R2)

**Comment by ansibot** [comment]
@TimothyVandenbrande, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by trondhindenes** [comment]
Pretty sure the firewall cmdlets were added in 2012R2.

**Comment by dreamcat4** [comment]
Right... when I glanced at the online docs pages somewhere yesterday, it seemed like the only solid requirement was 'PowerShell v3.0' ? Which (me not knowing much about it) kindda sounds pretty old now doesnt it?

Maybe I'm wrong about that (if yes then sorry). It just seemed to say that somewhere-or-other. Sorry I cannot check further that compatibility issues myself as only have shiny new win10 targets here.
